libtcmu_log.c

```diff
@@ -471,6 +483,100 @@ static void *log_thread_start(void *arg)
 	return NULL;
 }
 
+/* tcmu log dir path */
+static char *tcmu_log_dir = NULL;
+
+static bool tcmu_logdir_check(const char *path)
+{
+	if (!path)
+		return false;
+
+	if (strlen(path) > PATH_MAX - TCMU_LOG_FILENAME_MAX) {
+		tcmu_err("--tcmu-log-dir='%s' cannot exceed %d characters\n",
+			 path, PATH_MAX - TCMU_LOG_FILENAME_MAX);
+		return false;
+	}
+
+	return true;
+}
+
+/* get the log dir of tcmu-runner */
+char *tcmu_get_log_dir(void)
+{
+	return tcmu_log_dir;
+}
+
+static char *tcmu_alloc_and_set_log_dir(const char *log_dir)
+{
+	/*
+	 * Do nothing here and will use the /var/log/
+	 * as the default log dir
+	 */
+	if (!log_dir)
+		return NULL;
+
+	tcmu_log_dir = strdup(log_dir);
+	if (!tcmu_log_dir)
+		tcmu_err("Failed to copy log dir: %s\n", log_dir);
+
+	return tcmu_log_dir;
+}
+
+void tcmu_free_log_dir(void)
+{
+	if (tcmu_log_dir)
+		free(tcmu_log_dir);
+}
+
+static int tcmu_mkdirs(const char *pathname)
```

Comment:
@lxbsz 
#tcmu-runner --tcmu-log-dir="/var/log/tcmu-cli/"  --> works
#tcmu-runner --tcmu-log-dir="/var/log/tcmu-cli"   --> doesn't work

Note: last char '/'

Is this intentional ?

Reply:
Yes, thanks @pkalever, it is a bug.